src/components/global/Footer.jsx

```diff
@@ -1,8 +1,208 @@
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Container from '@mui/material/Container';
+import Link from '@mui/material/Link';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import { useTheme } from '@mui/material/styles';
+// Other Library Imports
+import dayjs from 'dayjs';
+import { FaDiscord, FaGithub, FaLinkedin, FaMeetup } from 'react-icons/fa6';
+// Image Imports
+import roseLogo from '/assets/rose_logo.png';
+import logoBlob from '/assets/logoBlob.svg';
+
+// Legal links
+const legalLinks = [
+  {
+    href: 'https://www.codepdx.org/',
+    title: 'About'
+  },
+  {
+    href: 'https://www.codepdx.org/',
+    title: 'Code of Conduct'
+  },
+  {
+    href: 'https://www.codepdx.org/',
+    title: 'Contact Us'
+  },
+  {
+    href: 'https://www.codepdx.org/',
+    target: '_blank',
+    rel: 'noopener',
+    ml: 0.5,
+    text: `©${dayjs().year()}`,
+    title: 'CODE PDX'
+  }
+];
+
+// Social media links
+const socialMediaLinks = [
+  {
+    href: 'https://github.com/codeforpdx/',
+    icon: <FaGithub size={45} />,
+    target: '_blank',
+    rel: 'noopener'
+  },
+  {
+    href: 'https://www.linkedin.com/company/code-pdx/',
+    icon: <FaLinkedin size={45} />,
+    target: '_blank',
+    rel: 'noopener'
+  },
+  {
+    href: 'https://www.meetup.com/Code-for-PDX/',
+    icon: <FaMeetup size={45} />,
+    target: '_blank',
+    rel: 'noopener'
+  },
+  {
+    href: 'https://discord.gg/x6b573et',
+    icon: <FaDiscord size={45} />,
+    target: '_blank',
+    rel: 'noopener'
+  }
+];
+
+// renders socialMediaLinks for placement in stack below
+const renderSocialLinks = socialMediaLinks.map(({ href, icon }) => (
+  <Link
+    key={href}
+    href={href}
+    target="_blank"
+    rel="noopener"
+    color="#000"
+    sx={{
+      '&:hover': {
+        color: 'secondary.main'
+      }
+    }}
+  >
+    {icon}
+  </Link>
+));
+
+const socialBlobStyle = {
+  backgroundImage: 'url(/assets/socialsBlob.svg)',
+  backgroundSize: 'cover',
+  backgroundRepeat: 'no-repeat',
+  backgroundPosition: 'center',
+  overflow: 'visible',
+  minWidth: '42%',
+  minHeight: '175px'
+};
+
+const logoBlobStyle = {
+  backgroundImage: 'url(/assets/logoBlob.svg)',
+  backgroundRepeat: 'no-repeat',
+  backgroundPosition: 'center',
+  backgroundSize: 'cover',
+  overflow: 'visible',
+  minHeight: '40vh',
+  minWidth: '40vh'
+};
+
 const Footer = () => {
-  // any logic needed
-  // return (
-  // all the JSX (looks like HTML) to render
-  // )
+  const theme = useTheme();
+
+  // maps legalLinks used for placement in stack below
+  const renderLegalLinks = legalLinks.map((link, index) => (
```

Comment:
why is this `.map()` here, while all others are about the `Footer` function/component?

Reply:
one of them uses the `useTheme()` hook and the other doesn't. Can't `useTheme()` outside of  react component so it had to be moved back into the footer. for now leaving renderSocialLinks up top because its less to scroll through while trying to get everything working but can be moved on final push.